src/Avalonia.Controls.TreeDataGrid/Automation/Peers/TreeDataGridRowAutomationPeer.cs

```diff
@@ -0,0 +1,21 @@
+using Avalonia.Automation.Peers;
+using Avalonia.Controls.Primitives;
+
+namespace Avalonia.Controls.Automation.Peers;
+
+public class TreeDataGridRowAutomationPeer : ControlAutomationPeer
+{
+    public TreeDataGridRowAutomationPeer(TreeDataGridRow owner)
+        : base(owner)
+    {
+    }
+
+    protected override AutomationControlType GetAutomationControlTypeCore()
+    {
+        return AutomationControlType.DataItem;
```

Comment:
It should be TreeItem, I think.
I was previously checking automation for Visual Studio PropertyGrid, and it was more behaving as a tree rather than grid.

Reply:
Row itself should also implement IValueProvider

Reply:
And possible IToggleProvider (since it can be expanded/collapsed)